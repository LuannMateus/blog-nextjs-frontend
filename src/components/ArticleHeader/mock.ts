export default {
  id: 'asdasdasdasd',
  title:
    'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
  excerpt:
    "Lorem Ipsum has been the industry's standard dummy text ever since the 1500s,",
  cover:
    'https://res.cloudinary.com/dlizakp2a/image/upload/v1614696630/Frozen_sunset_on_the_lake_by_Manuel_Arslanyan_9f9cd8ea10.jpg',
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    id: '123456789',
    slug: 'luan-mateus',
    displayName: 'Luan Mateus',
  },
  categories: [
    {
      id: 'asdslgyh65',
      displayName: 'Tech',
      slug: 'tech',
    },
    {
      id: 'asdslgyh66',
      displayName: 'JS',
      slug: 'javascript',
    },
  ],
};
