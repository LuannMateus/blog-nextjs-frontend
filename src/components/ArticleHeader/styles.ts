import styled, { css } from 'styled-components';
import { Title as HeadingStyle } from '../Heading/styles';

export const Wrapper = styled.section`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyle} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    margin: ${theme.spacings.medium} 0;
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    display: block;
    max-width: 100%;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
