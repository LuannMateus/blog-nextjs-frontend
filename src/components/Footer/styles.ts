import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    width: 100%;

    text-align: center;

    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
