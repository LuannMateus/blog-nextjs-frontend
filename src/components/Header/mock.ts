export default {
  blogName: 'Luan Mateus',
  blogDescription:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud.',
  logo: './assets/images/logo.jpg',
  showText: true,
};
