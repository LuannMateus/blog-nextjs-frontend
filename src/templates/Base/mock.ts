import { BaseTemplateProps } from '.';

export default {
  settings: {
    id: '61fd5d62845ac9392828c858',
    blogName: 'New Programming',
    blogDescription:
      'Esse blog é voltado para os amantes de tecnologia que querem aprender sempre um pouco mais.',
    logo: {
      id: 'some-id',
      alternativeText: 'New Programming',
      url: '/assets/images/logo.jpg',
    },
    menuLink: [
      {
        id: '61fd600506f225328cbe0647',
        link: 'https://pt-br.reactjs.org/docs/hooks-effect.html',
        children: 'useEffect',
        newTab: true,
      },
    ],
    footerHtml: 'A footer paragraph',
  },
  children: 'Olá mundo',
} as BaseTemplateProps;
