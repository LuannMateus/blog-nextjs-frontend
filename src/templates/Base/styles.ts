import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  padding-left: 4px;
`;

export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    padding-top: ${theme.spacings.xhuge};
  `}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;

    margin-bottom: ${theme.spacings.xhuge};

    @media ${theme.media.lteMedium} {
      width: 100%;
    }
  `}
`;

export const FooterContainer = styled.div`
  ${() => css`
    width: 100%;

    margin: 0 auto;
    padding: 0;
  `}
`;
