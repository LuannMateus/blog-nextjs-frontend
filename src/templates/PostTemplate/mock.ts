import { PostTemplateProps } from '.';

export default {
  settings: {
    id: '61fd5d62845ac9392828c858',
    blogName: 'New Programming',
    blogDescription:
      'Esse blog é voltado para os amantes de tecnologia que querem aprender sempre um pouco mais.',
    logo: {
      id: 'some-id',
      alternativeText: 'New Programming',
      url: '/assets/images/logo.jpg',
    },
    menuLink: [
      {
        id: '61fd600506f225328cbe0647',
        link: 'https://pt-br.reactjs.org/docs/hooks-effect.html',
        children: 'useEffect',
        newTab: true,
      },
    ],
    footerHtml: '<p>A footer paragraph</p>',
  },
  post: {
    id: '61fd5706925de013f4652773',
    slug: 'usando-effect-hook-hook-de-efeito',
    title: 'Usando Effect Hook (Hook de Efeito)',
    excerpt:
      'O Effect Hook (Hook de Efeito) te permite executar efeitos colaterais em componentes funcionais.',
    content:
      '<p><span style="color:rgb(0,0,0);">O </span><i>Effect Hook</i><span style="color:rgb(0,0,0);"> (Hook de Efeito) te permite executar efeitos colaterais em componentes funcionais:</span></p><pre><code class="language-plaintext">import React, { useState, useEffect } from \'react\';\nfunction Exemplo() {\n  const [count, setCount] = useState(0);\n\n  // Similar ao componentDidMount e componentDidUpdate:  useEffect(() =&gt; {    // Atualiza o titulo do documento usando a API do browser    document.title = `Você clicou ${count} vezes`;  });\n  return (\n    &lt;div&gt;\n      &lt;p&gt;Você clicou {count} vezes&lt;/p&gt;\n      &lt;button onClick={() =&gt; setCount(count + 1)}&gt;\n        Clique aqui\n      &lt;/button&gt;\n    &lt;/div&gt;\n  );\n}</code></pre><p>&nbsp;</p><p style="margin-left:0px;">Esse trecho de código é baseado no <a href="https://pt-br.reactjs.org/docs/hooks-state.html">exemplo de contador da página anterior</a>, mas nós adicionamos uma nova funcionalidade a ele: nós definimos o titulo do documento para ser uma mensagem customizada que inclua o número de cliques.</p><p style="margin-left:0px;">Buscar dados, configurar uma subscription, e mudar o DOM manualmente dentro dos componentes React são exemplos de efeitos colaterais. Esteja você acostumado ou não a chamar essas operações de “efeitos colaterais” (ou somente “efeitos”), você provavelmente já usou eles em seus componentes antes.</p><p style="margin-left:0px;">&nbsp;</p><p style="margin-left:0px;"><strong>Dica</strong></p><p style="margin-left:0px;">Se você está familiarizado com os métodos do ciclo de vida do React, você pode pensar no Hook <code>useEffect</code> como <code>componentDidMount</code>, <code>componentDidUpdate</code>, e <code>componentWillUnmount</code> combinados.</p><p style="margin-left:0px;">&nbsp;</p><p style="margin-left:0px;">Existem dois tipos comuns de efeitos colaterais nos componentes React: aqueles que não precisam de limpeza, e aqueles que precisam. Vamos ver as suas diferenças mais detalhadamente.</p><h2 style="margin-left:0px;">&nbsp;</h2><figure class="media"><oembed url="https://www.youtube.com/watch?v=jcc9T-5inrk"></oembed></figure>',
    allowComments: true,
    cover: {
      id: '61fd5673925de013f4652772',
      alternativeText: '',
      url: 'https://res.cloudinary.com/project-cdn/image/upload/v1643992684/use_Effect_hook_62405844a5.png',
    },
    categories: [
      {
        id: '61fd538bf6fd1926a0820a03',
        displayName: 'React',
        slug: 'react',
      },
    ],
    tags: [
      {
        id: '61fd5394f6fd1926a0820a04',
        displayName: 'useReducer',
        slug: 'use-effect',
      },
    ],
    author: {
      id: '61fd537ef6fd1926a0820a02',
      displayName: 'LuanMateus',
      slug: 'luan-mateus',
    },
  },
} as PostTemplateProps;
