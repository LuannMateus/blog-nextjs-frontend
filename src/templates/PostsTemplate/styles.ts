import styled, { css } from 'styled-components';

export const LoadMoreContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;

    display: flex;
    justify-content: center;
    align-items: center;

    padding-left: ${theme.spacings.medium};
    padding-top: ${theme.spacings.small};
  `}
`;
