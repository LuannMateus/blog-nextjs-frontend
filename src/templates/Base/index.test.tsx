import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { Base, BaseTemplateProps } from '.';

import mock from './mock';

const props: BaseTemplateProps = mock;

describe('<BaseTemplate />', () => {
  it('Should render base elements', () => {
    renderTheme(<Base {...props} />);

    expect(
      screen.getByRole('img', {
        name: 'New Programming - Esse blog Ã© voltado para os amantes de tecnologia que querem aprender sempre um pouco mais.',
      }),
    ).toBeInTheDocument();
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
    expect(screen.getByText(props.settings.text)).toBeInTheDocument();
    expect(screen.getByLabelText('Go to top')).toBeInTheDocument();
  });

  it('Should render base elements', () => {
    const { container } = renderTheme(<Base {...props} />);
    expect(container).toMatchSnapshot();
  });
});
