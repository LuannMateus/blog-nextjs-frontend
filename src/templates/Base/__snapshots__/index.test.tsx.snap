// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseTemplate /> Should render base elements 1`] = `
.c24 {
  width: 100%;
  text-align: center;
}

.c24 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
}

.c25 pre[class*='language-'],
.c25 code[class*='language-'] {
  color: #d4d4d4;
  font-size: 13px;
  text-shadow: none;
  font-family: Menlo,Monaco,Consolas,'Andale Mono','Ubuntu Mono','Courier New',monospace;
  direction: ltr;
  text-align: left;
  white-space: pre;
  word-spacing: normal;
  word-break: normal;
  line-height: 1.5;
  -moz-tab-size: 4;
  -o-tab-size: 4;
  tab-size: 4;
  -webkit-hyphens: none;
  -moz-hyphens: none;
  -ms-hyphens: none;
  -webkit-hyphens: none;
  -moz-hyphens: none;
  -ms-hyphens: none;
  hyphens: none;
}

.c25 pre[class*='language-']::selection,
.c25 code[class*='language-']::selection,
.c25 pre[class*='language-'] *::selection,
.c25 code[class*='language-'] *::selection {
  text-shadow: none;
  background: #75a7ca;
}

.c25 pre[class*='language-'] {
  padding: 1em;
  margin: 0.5em 0;
  overflow: auto;
  background: #1e1e1e;
}

.c25:not(pre) > code[class*='language-'] {
  padding: 0.1em 0.3em;
  border-radius: 0.3em;
  color: #db4c69;
  background: #f9f2f4;
}

.c25 .namespace {
  opacity: 0.7;
}

.c25 .token.doctype .token.doctype-tag {
  color: #569cd6;
}

.c25 .token.doctype .token.name {
  color: #9cdcfe;
}

.c25 .token.comment,
.c25 .token.prolog {
  color: #6a9955;
}

.c25 .token.punctuation,
.c25 .language-html .language-css .token.punctuation,
.c25 .language-html .language-javascript .token.punctuation {
  color: #d4d4d4;
}

.c25 .token.property,
.c25 .token.tag,
.c25 .token.boolean,
.c25 .token.number,
.c25 .token.constant,
.c25 .token.symbol,
.c25 .token.inserted,
.c25 .token.unit {
  color: #b5cea8;
}

.c25 .token.selector,
.c25 .token.attr-name,
.c25 .token.string,
.c25 .token.char,
.c25 .token.builtin,
.c25 .token.deleted {
  color: #ce9178;
}

.c25 .language-css .token.string.url {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c25 .token.operator,
.c25 .token.entity {
  color: #d4d4d4;
}

.c25 .token.operator.arrow {
  color: #569cd6;
}

.c25 .token.atrule {
  color: #ce9178;
}

.c25 .token.atrule .token.rule {
  color: #c586c0;
}

.c25 .token.atrule .token.url {
  color: #9cdcfe;
}

.c25 .token.atrule .token.url .token.function {
  color: #dcdcaa;
}

.c25 .token.atrule .token.url .token.punctuation {
  color: #d4d4d4;
}

.c25 .token.keyword {
  color: #569cd6;
}

.c25 .token.keyword.module,
.c25 .token.keyword.control-flow {
  color: #c586c0;
}

.c25 .token.function,
.c25 .token.function .token.maybe-class-name {
  color: #dcdcaa;
}

.c25 .token.regex {
  color: #d16969;
}

.c25 .token.important {
  color: #569cd6;
}

.c25 .token.italic {
  font-style: italic;
}

.c25 .token.constant {
  color: #9cdcfe;
}

.c25 .token.class-name,
.c25 .token.maybe-class-name {
  color: #4ec9b0;
}

.c25 .token.console {
  color: #9cdcfe;
}

.c25 .token.parameter {
  color: #9cdcfe;
}

.c25 .token.interpolation {
  color: #9cdcfe;
}

.c25 .token.punctuation.interpolation-punctuation {
  color: #569cd6;
}

.c25 .token.boolean {
  color: #569cd6;
}

.c25 .token.property,
.c25 .token.variable,
.c25 .token.imports .token.maybe-class-name,
.c25 .token.exports .token.maybe-class-name {
  color: #9cdcfe;
}

.c25 .token.selector {
  color: #d7ba7d;
}

.c25 .token.escape {
  color: #d7ba7d;
}

.c25 .token.tag {
  color: #569cd6;
}

.c25 .token.tag .token.punctuation {
  color: #808080;
}

.c25 .token.cdata {
  color: #808080;
}

.c25 .token.attr-name {
  color: #9cdcfe;
}

.c25 .token.attr-value,
.c25 .token.attr-value .token.punctuation {
  color: #ce9178;
}

.c25 .token.attr-value .token.punctuation.attr-equals {
  color: #d4d4d4;
}

.c25 .token.entity {
  color: #569cd6;
}

.c25 .token.namespace {
  color: #4ec9b0;
}

.c25 pre[class*='language-javascript'],
.c25 code[class*='language-javascript'],
.c25 pre[class*='language-jsx'],
.c25 code[class*='language-jsx'],
.c25 pre[class*='language-typescript'],
.c25 code[class*='language-typescript'],
.c25 pre[class*='language-tsx'],
.c25 code[class*='language-tsx'] {
  color: #9cdcfe;
}

.c25 pre[class*='language-css'],
.c25 code[class*='language-css'] {
  color: #ce9178;
}

.c25 pre[class*='language-html'],
.c25 code[class*='language-html'] {
  color: #d4d4d4;
}

.c25 .language-regex .token.anchor {
  color: #dcdcaa;
}

.c25 .language-html .token.punctuation {
  color: #808080;
}

.c25 pre[data-line] {
  position: relative;
}

.c25 pre[class*='language-'] > code[class*='language-'] {
  position: relative;
  z-index: 1;
}

.c25 .line-highlight {
  position: absolute;
  left: 0;
  right: 0;
  padding: inherit 0;
  margin-top: 1em;
  background: #f7ebc6;
  box-shadow: inset 5px 0 0 #f7d87c;
  z-index: 0;
  pointer-events: none;
  line-height: inherit;
  white-space: pre;
}

.c26 {
  width: 100%;
  font-size: calc(1.6rem + 0.2rem);
  line-height: 1.5;
}

.c26 iframe[src*='ads'] {
  display: none;
}

.c26 p {
  width: 100%;
  margin: 2.4rem 0;
  word-wrap: break-word;
}

.c26 a,
.c26 a:visited,
.c26 a:link {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c26 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c26 code {
  font-family: monospace;
  color: #dc143c;
  font-size: 1.6rem;
  background: #DDDDDD;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c26 pre {
  background: #000000;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFFFFF;
  margin: 2.4rem 0;
  overflow-x: auto;
  font-size: 1.6rem;
}

.c26 pre code {
  color: inherit;
  background: inherit;
}

.c26 img {
  max-width: 100%;
}

.c26 .image {
  background: #DDDDDD;
  line-height: 0;
  margin: 2.4rem 0;
}

.c26 .image figcaption {
  font-size: 1.6rem;
  padding: 1.6rem 0;
  text-align: center;
  line-height: 1.3;
  margin: 2.4rem 0;
}

.c26 .image-style-side {
  max-width: 50%;
  float: right;
  margin: 2.4rem;
}

.c26 hr {
  border: none;
  border: 1px solid red;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c26 ul,
.c26 ol {
  margin: 2.4rem 4.0rem;
}

.c26 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c26 table {
  width: 100%;
  border-collapse: collapse;
}

.c26 table td,
.c26 table th {
  padding: 1.6rem;
  border: 0.1rem solid #DDDDDD;
}

.c26 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #AAAAAA;
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

.c8 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c27 {
  position: fixed;
  background: #000000;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
}

.c11 {
  line-height: 1.2;
  color: #000000;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c19 {
  line-height: 1.2;
  color: #000000;
  font-size: 2.4rem;
  text-transform: none;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c13 > img {
  width: 15rem;
  height: 15rem;
  border-radius: 50%;
}

.c17 {
  max-width: 96rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 1.6rem;
  padding-bottom: 6.4rem;
  color: #333333;
  font-size: 1.6rem;
  margin: 0 auto;
}

.c17 .c10 {
  margin-right: calc(1.6rem - 1rem);
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  margin-left: 2.4rem;
}

.c15 {
  display: block;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  border-right: 0.5rem solid #000000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  cursor: pointer;
}

.c15:hover {
  border-right: 0.5rem solid #dc143c;
  color: #dc143c;
}

.c6 {
  width: 27rem;
  height: 100%;
  background: #000000;
  border-top-right-radius: 4px;
  padding: 0.8rem calc(0.8rem - 0.6rem);
  position: fixed;
  left: -26rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  z-index: 1001;
}

.c6 .c12 {
  margin-top: 6.4rem;
}

.c6 .c12 img {
  border: 0.5rem solid #dc143c;
}

.c6.active {
  left: 0rem;
  overflow-y: auto;
  overflow-x: hidden;
}

.c9 {
  width: 100%;
  max-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  background-color: #000000;
  border-radius: 4px;
  left: 2rem;
}

.c7 svg {
  width: 3rem;
  color: #FFFFFF;
  padding-right: 3.5rem;
  padding-top: 1rem;
  cursor: pointer;
  -webkit-transition: filter 300ms ease-in-out;
  transition: filter 300ms ease-in-out;
}

.c7 svg:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c7 .menuIcon {
  padding: 0;
}

.c14 {
  width: 90%;
  margin: 4rem 0;
}

.c1 {
  position: fixed;
  top: 2rem;
  right: 2rem;
  z-index: 2;
}

.c2 {
  position: relative;
  display: inline-block;
  width: 5rem;
  height: 2.5rem;
  line-height: 0;
  font-size: 0;
  overflow: hidden;
  color: rgba(0,0,0,0);
}

.c3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  width: 0;
  height: 0;
}

.c3:checked + .c4 {
  background: #dc143c;
}

.c3:focus + .c4 {
  box-shadow: 0 0 1px #dc143c;
}

.c3:checked + .c4:before {
  -webkit-transform: translateX(1.9rem);
  -ms-transform: translateX(1.9rem);
  transform: translateX(1.9rem);
  background: #000000;
}

.c5 {
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: #DDDDDD;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  border-radius: 2rem;
  box-shadow: 0 0 1px #000000;
}

.c5:before {
  content: '';
  position: absolute;
  height: 2.1rem;
  width: 2.1rem;
  left: 0.4rem;
  bottom: 0.25rem;
  background: #333333;
  border-radius: 50%;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 svg.search-ok-icon,
.c0 svg.search-cancel-icon {
  width: 2.4rem;
  height: 2.4rem;
  margin-left: 1rem;
}

.c0 svg.search-cancel-icon {
  color: #dc143c;
}

.c16 {
  padding-top: 6.4rem;
  padding-left: 20px;
}

.c20 {
  margin: 0 auto;
  margin-bottom: 4.0rem;
  padding: 0 3.2rem;
  max-width: 120rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c21 {
  padding: 0.5rem 1.6rem;
}

.c21:disabled {
  outline: 0.1rem solid #dc143c;
}

.c22 {
  max-width: 120rem;
  width: 100%;
  margin-bottom: 6.4rem;
}

.c23 {
  width: 100%;
  margin: 0 auto;
  padding: 0;
}

@media print {
  .c25 pre[class*='language-'],
  .c25 code[class*='language-'] {
    text-shadow: none;
  }
}

@media (max-width:768px) {
  .c26 {
    font-size: 2rem;
  }

  .c26 pre {
    width: 82%;
  }

  .c26 .image-style-side {
    max-width: 100%;
    float: none;
    margin: 0;
  }
}

@media (max-width:768px) {
  .c17 {
    margin: 0;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:768px) {
  .c18 {
    text-align: center;
    margin-top: 1.6rem;
  }
}

@media (max-width:768px) {
  .c6 {
    width: 25.5rem;
  }
}

@media (max-width:768px) {
  .c22 {
    width: 100%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        aria-label="Toggle light and dark modes"
        class="c2"
      >
        <input
          class="c3"
          type="checkbox"
          value="Toggle light and dark modes"
        />
        <span
          class="c4 c5"
        />
      </label>
    </div>
    <div
      class="c6"
    >
      <div
        class="c7"
      >
        <svg
          aria-label="Open menu"
          class="c8 menuIcon"
          fill="currentColor"
          focusable="false"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Open menu
          </title>
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </div>
      <nav
        aria-hidden="true"
        class="c9"
      >
        <h1
          class="c10 c11"
        >
          <a
            class="c12 c13"
            href="/"
            target="_self"
          >
            <img
              alt="New Programming"
              src="/assets/images/logo.jpg"
            />
          </a>
        </h1>
        <section
          class="c14"
        >
          <a
            class="c15"
            href="https://pt-br.reactjs.org/docs/hooks-effect.html"
            target="_blank"
          >
            useEffect
          </a>
        </section>
      </nav>
    </div>
    <div
      class="c16"
    >
      <header
        class="c17"
      >
        <h1
          class="c10 c11"
        >
          <a
            class="c12 c13"
            href="/"
            target="_self"
          >
            <img
              alt="New Programming - Esse blog é voltado para os amantes de tecnologia que querem aprender sempre um pouco mais."
              src="/assets/images/logo.jpg"
            />
          </a>
        </h1>
        <section
          class="c18"
        >
          <h1
            class="c10 c19"
          >
            New Programming
          </h1>
          <p>
            Esse blog é voltado para os amantes de tecnologia que querem aprender sempre um pouco mais.
          </p>
        </section>
      </header>
    </div>
    <div
      class="c20"
    >
      <input
        class="c21"
        name="q"
        placeholder="Encontre posts"
        type="search"
        value=""
      />
      <svg
        aria-hidden="true"
        aria-label="Input enabled"
        class="c8 search-ok-icon"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm4.59-12.42L10 14.17l-2.59-2.58L6 13l4 4 8-8z"
        />
      </svg>
    </div>
    <div
      class="c22"
    >
      Olá mundo
    </div>
    <div
      class="c23"
    >
      <footer
        class="c24"
      >
        <div
          class="c25"
        >
          <div
            class="c26"
          >
            A footer paragraph
          </div>
        </div>
      </footer>
    </div>
    <a
      aria-label="Go to top"
      class="c27"
      href="#"
      title="Go to top"
    >
      <svg
        aria-hidden="true"
        class="c8"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
        />
      </svg>
    </a>
  </div>
</div>
`;
